### ISSUE TITLE
Unable to delete things.

### ISSUE THREAD

**Issue by mortimus1987** [jovandeginste/workout-tracker#321]
I have uploaded a brief test track, to check the import functionality from my phone. Having it confirmed it worked, I attempted to delete the workout but received the following error:

Something went wrong: ERROR: update or delete on table "map_data" violates foreign key constraint "fk_map_data_details" on table "map_data_details" (SQLSTATE 23503)

I get the a similar error when attempting to delete the initial admin user (after creating another user and ensuring they have admin privileges) but for "fk_users_profile" on table "profiles".

I am using a docker compose deployment with PostGres 16 alpine as the database.

**Comment by jovandeginste** [comment]
Can you check the exact version you are running?

**Comment by mortimus1987** [comment]
The docker image is the :latest tag. Workout tracker itself reports branch master (c6d4ea03)

**Comment by mortimus1987** [comment]
No; using sqlite I seem to be able to delete both workouts and users with no issue, so that suggests its PostGres related.

Interestingly, initially the docker container kept crashing when I tried to use sqlite instead of PG. After checking the logs I realised this was because my JWT environment variable was a mess with brackets and spaces etc. (failure of variable substitution by ansible when I deployed the compose stack because of my fat fingers).  I don't know if thats expected or not, but using the exact same env var the container started fine when using postgres.

Just to be clear, I corrected the JWT var and made a new PG database, and the inability to delete persists, so its not related, but thought it worth mentioning in case its also unintentional behaviour.

**Comment by jovandeginste** [comment]
Then I will need to test with a postgres backend

**Comment by mortimus1987** [comment]
Using the latest docker container (6d657465) and I have been able to delete the administrator user successfully. However, attempting to delete the workout I uploaded as a test still fails with the same error.

**Comment by jovandeginste** [comment]
> Using the latest docker container ([6d65746](https://github.com/jovandeginste/workout-tracker/commit/6d657465d5a00b14499645972269afd0d75d35e1)) and I have been able to delete the administrator user successfully. However, attempting to delete the workout I uploaded as a test still fails with the same error.

Was this a new test, or the previous database content?

**Comment by jovandeginste** [comment]
Can you make sure to use the latest image (from commit e55abb5b6820e48817f953af637eb4c6446d3781)?

**Comment by sikmir** [comment]
> Can you make sure to use the latest image (from commit [e55abb5](https://github.com/jovandeginste/workout-tracker/commit/e55abb5b6820e48817f953af637eb4c6446d3781))?

Yes, I use the latest one with postgresql backend.

**Comment by jovandeginste** [comment]
Could you find a way to reproduce? I tried here but to no avail...

**Comment by sikmir** [comment]
> Could you find a way to reproduce? I tried here but to no avail...

It's easy to reproduce, since it happens each time I try to delete a workout. What additional info can help here?

**Comment by jovandeginste** [comment]
Can you reproduce from an empty setup? Because I can not reproduce from scratch.

**Comment by sikmir** [comment]
> Can you reproduce from an empty setup? Because I can not reproduce from scratch.

It is new setup, I've deployed it to the server and just added some tracks.

**Comment by sikmir** [comment]
I've just checked with 2.0.3, there is no error now, but still no way to delete workouts.

**Comment by jovandeginste** [comment]
I will try to reproduce in my test setup... So far no luck.

**Comment by sikmir** [comment]
I've finally dropped old database and just recreated a new one with 2.0.3. And deleting things work now!

**Comment by jovandeginste** [comment]
Well, at least that's fixed for you. I'll try to add more tests to the suite, try to catch these issues before they get deployed...
